NE-2076: Add dynamic commit for operand image

Thealisyed opened the pull request:
Added the FULL_COMMIT to the
tekton pipeline and Containerfile in order
to have dynamic commit for the operand
image


**Description**

Added the FULL_COMMIT to the
tekton pipeline and Containerfile in order
to have dynamic commit for the operand
image

**Checklist**

- [NA ] Unit tests updated
- [NA ] End user documentation updated

Commit:
NE-2076: Add dynamic commit for operand image

Added the FULL_COMMIT  to the
tekton pipeline and Containerfile in order
to have dynamic commit for the operand
image

alebedev87 commented on Containerfile.externaldns:
```diff
@@ -29,7 +29,9 @@ LABEL maintainer="Red Hat, Inc."
-LABEL commit="76d92ad82b22c92c191a8c0145d3712e4012d987"
+LABEL io.openshift.build.commit.id=${FULL_COMMIT}
+LABEL io.openshift.build.source-location=${GIT_URL}
+LABEL io.openshift.build.commit.url=${GIT_URL}/commit/${FULL_COMMIT}
```
`commit.id` is a label which was used for the bundle container image. `commit` is what's used for this image, let's not break the format (just yet, we can rethink it later if there will be a request). Here are all the labels of this image from [the latest CPaaS build](https://brewweb.engineering.redhat.com/brew/buildinfo?buildID=3833646):
```
$ podman inspect registry-proxy.engineering.redhat.com/rh-osbs/external-dns:1.3.1-3 | jq .[0].Config.Labels
{
  "architecture": "x86_64",
  "build-date": "2025-09-08T10:19:32",
  "com.redhat.component": "external-dns-container",
  "com.redhat.license_terms": "https://www.redhat.com/agreements",
  "commit": "8da2509b922d50ef7b1b8ea2297758888f32448d",
  "description": "The Universal Base Image is designed and engineered to be the base layer for all of your containerized applications, middleware and utilities. This base image is freely redistributable, but Red Hat only supports Red Hat technologies through subscriptions for Red Hat products. This image is maintained by Red Hat and updated regularly.",
  "distribution-scope": "public",
  "io.buildah.version": "1.33.12",
  "io.k8s.description": "The Universal Base Image is designed and engineered to be the base layer for all of your containerized applications, middleware and utilities. This base image is freely redistributable, but Red Hat only supports Red Hat technologies through subscriptions for Red Hat products. This image is maintained by Red Hat and updated regularly.",
  "io.k8s.display-name": "Red Hat Universal Base Image 9",
  "io.openshift.expose-services": "",
  "io.openshift.tags": "base rhel9",
  "maintainer": "Red Hat, Inc.",
  "name": "external-dns",
  "release": "3",
  "summary": "Provides the latest release of Red Hat Universal Base Image 9.",
  "url": "https://access.redhat.com/containers/#/registry.access.redhat.com/external-dns/images/1.3.1-3",
  "vcs-ref": "bb33a8606f5ad38a05c13f2bfbcfbd83568aa7b8",
  "vcs-type": "git",
  "vendor": "Red Hat, Inc.",
  "version": "1.3.1"
}
```

Can you update the version to `1.3.1` along the way?

alebedev87 commented on Containerfile.externaldns:
```diff
@@ -28,8 +28,8 @@ FROM registry.access.redhat.com/ubi9/ubi:9.6-1752625787
-LABEL version="1.3.0"
-LABEL commit="76d92ad82b22c92c191a8c0145d3712e4012d987"
+LABEL version="1.3.1"
+LABEL commit=${FULL_COMMIT}
```
You pass `FULL_COMMIT` as a building arg however it's not set as `ARG` for this Containerfile.

alebedev87 reviewed (commented):
You removed `GIT_URL` from the build args but did update the commit message and PR description.

alebedev87 commented on .tekton/external-dns-container-ext-dns-optr-1-3-rhel-9-pull-request.yaml:
We should update the push pipeline too. Btw we forgot to do it in [the operator repository](https://github.com/openshift/external-dns-operator/pull/272) (something to fix there too).

alebedev87 commented on .tekton/external-dns-container-ext-dns-optr-1-3-rhel-9-pull-request.yaml:
```diff
@@ -212,6 +212,7 @@ spec:
       - name: BUILD_ARGS
         value:
         - $(params.build-args[*])
+        - 'FULL_COMMIT=$(tasks.clone-repository.results.short-commit)'
```
```suggestion
        - 'FULL_COMMIT=$(tasks.clone-repository.results.commit)'
```

Just noticed that we made the same mistake in the operator repository.

alebedev87 reviewed (commented):
I don't see the commit label value in the artifact built by Konflux:
```
$ podman inspect quay.io/redhat-user-workloads/external-dns-operator-tenant/ext-dns-optr-1-3-rhel-9/external-dns-container-ext-dns-optr-1-3-rhel-9:on-pr-d7f4f68dd4cec5e74444c5a5e225479a3e95403d | jq .[0].Config.Labels.commit
""
```

Can you?

alebedev87 commented:
/assign

Thealisyed commented:
```
➜  podman inspect --format '{{json .Config.Labels}}' quay.io/redhat-user-workloads/external-dns-operator-tenant/ext-dns-optr-1-3-rhel-9/external-dns-container-ext-dns-optr-1-3-rhel-9:on-pr-41d8ccf10d4e9b941ef86c7d93a07f6290d53757 | jq .
{
  "architecture": "x86_64",
  "build-date": "2025-10-02T15:41:34",
  "com.redhat.component": "external-dns-container",
  "com.redhat.license_terms": "https://www.redhat.com/en/about/red-hat-end-user-license-agreements#UBI",
  "commit": "41d8ccf10d4e9b941ef86c7d93a07f6290d53757",
  "description": "The Universal Base Image is designed and engineered to be the base layer for all of your containerized applications, middleware and utilities. This base image is freely redistributable, but Red Hat only supports Red Hat technologies through subscriptions for Red Hat products. This image is maintained by Red Hat and updated regularly.",
  "distribution-scope": "public",
  "io.buildah.version": "1.41.0-dev",
  "io.k8s.description": "The Universal Base Image is designed and engineered to be the base layer for all of your containerized applications, middleware and utilities. This base image is freely redistributable, but Red Hat only supports Red Hat technologies through subscriptions for Red Hat products. This image is maintained by Red Hat and updated regularly.",
  "io.k8s.display-name": "Red Hat Universal Base Image 9",
  "io.openshift.expose-services": "",
  "io.openshift.tags": "base rhel9",
  "maintainer": "Red Hat, Inc.",
  "name": "external-dns",
  "quay.expires-after": "5d",
  "release": "1752625787",
  "summary": "Provides the latest release of Red Hat Universal Base Image 9.",
  "url": "https://www.redhat.com",
  "vcs-ref": "41d8ccf10d4e9b941ef86c7d93a07f6290d53757",
  "vcs-type": "git",
  "vendor": "Red Hat, Inc.",
  "version": "1.3.1"
}
als
```

alebedev87 commented:
/lgtm
/approve
